Adding missing Configurations.props for refs and fixing the existing ones

joperezr opened the pull request:
cc: @ericstj @tarekgh @weshaggard 
FYI: @chcosta 

 - Fixing the few Configurations.props we have for refs to remove their claimed `netstandard` config.
 - Adding uap configuration to all refs.
 - Adding configuration file to refs that didn't have one already.
 - Fixed uap logic on targetGroups in order to select the right imports.

joperezr commented:
@tarekgh With these changes, building `src\ref.builds` with `/p:TargetGroup=netfx` will skip all of the reference assemblies. `build.cmd -framework:netfx` will not work yet, because it still will try to build some configurations of src\src.builds and have a bunch of issues.

joperezr commented:
@weshaggard let me know if these changes are more or less what you had in mind, and if so I'll go ahead and start working on adding the right configurations to the implementation projects.

tarekgh commented:
LGTM

weshaggard commented on src/System.Collections.Concurrent/ref/Configurations.props:
```diff
@@ -2,8 +2,8 @@
     <BuildConfigurations>
-      netstandard;
       netcoreapp;
+      uap;
```
With these we should also clean out the netcoreapp #ifdef's from the project and source files.

weshaggard commented:
Yes this is along the lines of what I was expecting. You should also go through and clean out the #ifdefs.

@ericstj should we add a default define for each configuration so it is present for each build config automatically?

joperezr commented:
> You should also go through and clean out the #ifdefs. 

Done with 904aa39

chcosta commented on binplace.targets:
```diff
@@ -35,6 +35,9 @@
       <TargetingPackDir Include="$(RefPath)" />
       <TargetingPackDir Condition="'$(IsNETCoreAppRef)' == 'true'" Include="$(BinDir)netcoreapp\pkg\ref" />
     </ItemGroup>
+    <ItemGroup Condition="'$(NuGetTargetMoniker)' == '' Or $(NuGetTargetMoniker.Contains('UAP'))">
```
Just to clarify, if `'$(NuGetTargetMoniker)' == ''`, then the TargetingPackDir includes both `$(RefPath)` and `$(RefRootPath)uap/`?

weshaggard replied:
RefPath should already be set to `$(RefRootPath)uap` so we shouldn't need to add this. Also this is going to cause issues for tests which blindly take *.dll from all the TargetingPackDir's

joperezr replied:
`'$(NuGetTargetMoniker)' == ''` part of the condition was actually copy/pasted by mistake, the only condition that should be there is the UAP one. @weshaggard RefPath is not set to UAP since in dir.props it gets initialized to netcoreapp, so perhaps the change has to go there instead. I'll remove this small change on the file and try to figure out a better way to calculate this.

weshaggard replied:
Ah yes I forget we hardcoded that. We do need to move to a place where we compute that variable based on the BuildConfiguration.

ericstj replied:
BuildConfiguration or Configuration?  I'd imagine that it should be based off of Configuration.

weshaggard replied:
No this one should be based on BuildConfiguration, as we want to consistently create the targeting pack folders per BuildConfiguration.

ericstj replied:
I think that's broken.  Consider I build a project with only a netstandard configuration for both BuildConfiguration netcoreapp and uap: separate vertical builds on same machine.  It will reuse the same intermediate/bin directories (per design) but with different refs we'll potentially have different assemblies.

weshaggard replied:
I'm not sure I follow. It will be the same assembly just loaded from a different ref directory. The entire idea is that we want an easy reference resolution from flat targeting pack for each vertical BuildConfiguration and not have to try and add a bunch of logic to pick the correct reference for every library. We aren't in the contract store or nuget resolution world when we build a vertical. To resolve a Reference you are only looking in one targeting pack directory.

ericstj commented on src/System.Net.Http.WinHttpHandler/ref/Configurations.props:
```diff
@@ -0,0 +1,9 @@
+  <PropertyGroup>
+    <BuildConfigurations>
+      netcoreapp;
+      uap;
```
I don't think this one is supported on UAP.

joperezr replied:
I believe it is, by looking at some pre-dev/eng packages it seems to have a netstandard1.3 reference assembly.

joperezr replied:
fixed.

ericstj commented on src/System.Net.HttpListener/ref/Configurations.props:
```diff
@@ -0,0 +1,9 @@
+  <PropertyGroup>
+    <BuildConfigurations>
+      netcoreapp;
+      uap;
```
Supported on UAP?

weshaggard replied:
What do you think our best way to figure out uap support?

ericstj replied:
Look at existing pkgproj.

ericstj commented on src/System.ValueTuple/ref/Configurations.props:
```diff
@@ -1,10 +1,9 @@
     <BuildConfigurations>
       netcoreapp;
-      netstandard1.0;
-      <!-- portable_net40+sl4+win8+wp8-Windows_NT; Looks like our filtering mechanism doesn't understand this targetgroup. Do we even need it? -->
```
Yes we need it.  Please add to data model.

ericstj commented on src/System.ValueTuple/ref/Configurations.props:
```diff
@@ -1,10 +1,9 @@
   <PropertyGroup>
     <BuildConfigurations>
       netcoreapp;
-      netstandard1.0;
```
We need to preserve the older configs of this one.  It'll ship a package.

ericstj replied:
... And on desktop I'd expect it to build the netstandard1.0 config.

ericstj commented on src/System.ValueTuple/ref/System.ValueTuple.csproj:
```diff
@@ -27,7 +27,7 @@
     <TargetingPackReference Include="System.Core" />
   </ItemGroup>
-  <ItemGroup Condition="'$(TargetGroup)' == 'netcoreapp'">
+  <ItemGroup>
```
Don't remove this, update the condition.

ericstj commented on src/Tools/GenerateProps/targetgroups.props:
```diff
@@ -13,6 +13,22 @@
+      <Imports>netcore50aot</Imports>
+      <CompatibleWith>netstandard2.0</CompatibleWith>
+    </TargetGroups>
+    <TargetGroups Include="uap">
```
Should we also add a uapaot?

weshaggard replied:
Probably and we are going to need some corert variants as well.

joperezr commented:
@weshaggard @ericstj I've addressed the feedback and removed the uap configurations for the projects where we didn't support uwp previously. Also, I've added Eric's new logic for calculating RefPath based on the original TargetGroup

weshaggard commented on src/System.DirectoryServices/ref/Configurations.props:
```diff
@@ -2,8 +2,7 @@
     <BuildConfigurations>
-      netcoreapp;
-      uap;
+      ;
```
Is this one intentionally empty?

joperezr replied:
yes. We haven't really added anything to these `DirectoryServices` assemblies yet, so it makes no sense to have them building on any framework. Plus, the code on the implementation side that we do have doesn't compile yet, so build is disabled there as well.
